app/views/users/index.html.erb

```diff
@@ -4,8 +4,38 @@
       <li>
         Name: <%= user.name %>
         <span class="profile-link">
-          <%= link_to 'See Profile',  user_path(user), class: 'profile-link' %>
+          <%= link_to 'See Profile', user_path(user), class: 'profile-link', id: "user-#{user.id}" %>
         </span>
+        <% if @logged_user.friend_request_sent?(user) %>
+          <span class="friendship-status" id=<%="waiting-user-#{user.id}"%> >Friend Request Sent</span>
+        <% elsif @logged_user.friend_request_received?(user) %>
+          <span class="friendship-status">Friendship Invitation Received:</span>
+          <% correct_friendship_id =
+                 Friendship.get_friendship_id(@all_friendships, @logged_user.id, user.id) %>
```

Comment:
- [ ] There should be NO application logic in the views nor helpers (no Post.new, Comment.create, current_user.friends.find_by(...)...). Everything connected with data processing should be encapsulated in models and used in controllers.

![](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/section/no_logic_in_helpers_requirement.png)

Reply:
It was refactored as requested, thank you very much @mosaaleb